src/builders.rs

```diff
@@ -1230,3 +1233,129 @@ impl From<DateBased> for Output {
         })))
     }
 }
+
+/// A generic formatter that easily provides good defaults while being configurable
+pub struct FormatterBuilder {
+    #[cfg(feature = "colored")]
+    color_config: Option<ColoredLevelConfig>,
+    #[cfg(feature = "chrono")]
+    chrono: bool,
+    level: bool,
+    target: bool,
+}
```

Comment:
I'm all for this, and we can definitely keep it here for now, but I'd rather the final "refactor" PR _not_ include any new features. Ideally I'd be able to merge one PR which refactors everything but keeps the exact same feature set, and then review and merge future PRs which depend on that refactoring separately.